### DISCUSSION TITLE
snmp examples of new field 'thresholds to performance data' in oid-files

### DISCUSSION THREAD

**Discussion by mdicss** [Linuxfabrik/monitoring-plugins#768] [Q&A]
Hi all
The snmp check changed in the last release of the plugins. Can someone give an explanation, what the new field 'thresholds to performance data' in oid-files exactly should do. It would be helpful to have some more examples with oid-files containing values for this field too.

**Comment by markuslf** [comment]
Should be already described here: https://github.com/Linuxfabrik/monitoring-plugins/tree/main/check-plugins/snmp

**Reply by mdicss** [reply]
Sorry, but for me, the description is a bit to short. I did a series of tests too and did not really find out, what this property does and what the difference to the warn and crit fields is.
But you'r right, now I see what the field does. For me, this is a bit surprising, as you may define other values there, that do not correspond to the used warn and critical thresholds defined in the other fields.

**Comment by markuslf** [comment]
Please feel free to improve the documentation from your perspective. We always welcome suggestions for improvement.

**Reply by mdicss** [reply]
ok. It would be nice to have an option, to write the defined values for warn and critical in the performance data instead of repeating them or even write some other values in the performance data.

**Comment by markuslf** [comment]
The README now documents this field in detail. The key point: the WARN and CRIT columns are Python *conditions* (e.g. `value > 80`), not numbers, so they cannot be written into the perfdata automatically. "Perfdata Alert Thresholds" holds the numeric thresholds for the perfdata only and has no effect on the state. That is why the values can differ from WARN/CRIT and why you normally just repeat them.

Auto-deriving the perfdata thresholds from WARN/CRIT is not feasible in general, since those conditions can be arbitrary expressions (including ones that span several OIDs) with no single numeric equivalent.
